File: disc01/disc01.java
public static List<Integer> evens(List<Integer> L) {
    List<Integer> list = new ArrayList<>();
    for (int num : L) {
        if (num % 2 == 0) {
            list.add(num);
        }
    }
    return list;
}

public static Map<String, Integer> countWords(List<String> words) {
    Map<String, Integer> map = new TreeMap<>();
    for (String word : words) {
        if (map.containsKey(word)) {
            map.put(word, map.get(word) + 1);
        } else {
            map.put(word, 1);
        }
    }
    return map;
}

File: disc01/disc01.py
def evens(list_of_ints):
    """Returns a copy of the list but keeping only the even numbers."""
    return_list = []
    for num in list_of_ints:
        if num % 2 == 0:
            return_list.append(num)
    return return_list


def count_words(list_of_words):
    """Returns a map from each word to its count."""
    counts = {}


class Dog:
    def __init__(self, name, size):
        self.name = name
        self.size = size

    def __repr__(self):
        return "Dog({}, {})".format(self.name, self.size)

    
dogs = [Dog("maya", 1000), Dog("yipster", 5), Dog("scott", 25)]
print(dogs[0])

